### PR TITLE
Set tls options for external proxies

### PR DESCRIPTION
**Pull request by timgerstel**
Signed-off-by: Timothy Gerstel <tgerstel@rocketsoftware.com>

## Proposed changes

This PR addresses Issue: [*Link to Github issue within https://github.com/zowe/zlux/issues* if any]

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made
- [ ] Relevant update to CHANGELOG.md
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

**Review by DivergentEuropeans — APPROVED** [review]
This removes an argument for something that we defaulted to false prior and we're adding tls options, so this should be okay

File: lib/index.js [hunk 1]
```diff
@@ -305,7 +305,7 @@ Server.prototype = {
       }
     }
 
-    const proxiedOptions = util.getAgentRequestOptions(this.userConfig, webAppOptions.tlsOptions, false);
```
Comment (1000TurquoisePogs) [inline lib/index.js]: don't remove arguments.

File: lib/util.js [hunk 2]
```diff
@@ -138,9 +138,7 @@ module.exports.getAgentRequestOptions = function(serverConfig, tlsOptions, inclu
     if ((typeof tlsOptions == 'object') && isHttps) {
       options = Object.assign(options, tlsOptions);
       delete options.key;
-      if (!includeCert) {
```
Comment (1000TurquoisePogs) [inline lib/util.js]: keep this. just use true/false if you need to do something.

**Review by 1000TurquoisePogs — CHANGES_REQUESTED** [review]

**Comment by DivergentEuropeans** [comment]
Also needs CHANGELOG

**Review by 1000TurquoisePogs — APPROVED** [review]

**Pull request merged**